trestle/tasks/csv_to_oscal_cd.py

```diff
@@ -506,26 +537,29 @@ def _create_rule_props(self, rule_key: tuple) -> List[Property]:
         rule_set = self._rule_set_id_mgr.get_next_rule_set_id()
         row_number = self._csv_mgr.get_row_number(rule_key)
         rule_set_mgr = _RuleSetMgr(row_number, rule_set)
-        column_names = CsvColumn.get_filtered_required_column_names() + CsvColumn.get_filtered_optional_column_names()
         namespace = self._get_namespace(rule_key)
+        if self._is_validation(rule_key):
+            column_names = CsvColumn.get_check_property_column_names()
+        else:
+            column_names = CsvColumn.get_rule_property_column_names()
         # req'd & optional props
         for column_name in column_names:
             prop_name = self._get_prop_name(column_name)
             prop_value = self._csv_mgr.get_value(rule_key, column_name).strip()
             rule_set_mgr.add_prop(prop_name, prop_value, namespace, self.get_class(prop_name))
-        # parameter columns
-        column_names = CsvColumn.get_parameter_column_names()
-        for column_name in column_names:
-            prop_name = self._get_prop_name(column_name)
-            prop_value = self._csv_mgr.get_value(rule_key, column_name).strip()
-            rule_set_mgr.add_prop(prop_name, prop_value, namespace, self.get_class(prop_name))
-        # user props
-        column_names = self._csv_mgr.get_user_column_names()
-        for column_name in column_names:
-            prop_name = self._get_prop_name(column_name)
-            prop_value = self._csv_mgr.get_value(rule_key, column_name).strip()
-            rule_set_mgr.add_prop(prop_name, prop_value, namespace, self.get_class(prop_name))
-        rule_set_mgr.validate()
+        if not self._is_validation(rule_key):
+            # parameter columns
+            column_names = CsvColumn.get_parameter_column_names()
+            for column_name in column_names:
+                prop_name = self._get_prop_name(column_name)
+                prop_value = self._csv_mgr.get_value(rule_key, column_name).strip()
+                rule_set_mgr.add_prop(prop_name, prop_value, namespace, self.get_class(prop_name))
+            # user props
+            column_names = self._csv_mgr.get_user_column_names()
+            for column_name in column_names:
+                prop_name = self._get_prop_name(column_name)
+                prop_value = self._csv_mgr.get_value(rule_key, column_name).strip()
+                rule_set_mgr.add_prop(prop_name, prop_value, namespace, self.get_class(prop_name))
         return rule_set_mgr.get_props()
```

Comment:
@degenaro Are user columns only getting added for non-validation component? I believe it should be added for all components.

Reply:
Fixed.